app/controllers/register-user.controller.js

```diff
@@ -158,18 +158,28 @@ const submitOtpVerify = async function submitOtpVerify (req, res, next) {
   }
 
   try {
-    const user = await registrationService.verifyOtpAndCreateUser(sessionData.code, verificationCode, correlationId)
-    loginController.setupDirectLoginAfterRegister(req, res, user.external_id)
-    return res.redirect(303, paths.registerUser.logUserIn)
+    await registrationService.verifyOtp(sessionData.code, verificationCode, correlationId)
   } catch (err) {
     if (err.errorCode === 401) {
       sessionData.recovered = {
         errors: {
           verificationCode: 'The verification code you’ve used is incorrect or has expired'
         }
       }
-      res.redirect(303, paths.registerUser.otpVerify)
+      return res.redirect(303, paths.registerUser.otpVerify)
     } else if (err.errorCode === 410) {
+      return renderErrorView(req, res, EXPIRED_ERROR_MESSAGE, 410)
+    } else {
+      return next(err)
+    }
+  }
+
+  try {
+    const completeResponse = await registrationService.completeInvite(sessionData.code, correlationId)
+    loginController.setupDirectLoginAfterRegister(req, res, completeResponse.user_external_id)
+    return res.redirect(303, paths.registerUser.logUserIn)
+  } catch (err) {
+    if (err.errorCode === 410) {
```

Comment:
Looks like we don't have tests that cover the various error responses from adminusers which we handle differently. I'd be wary to add new functional tests for this. Maybe this is a case where we could start writing some unit tests for this class to add this particular coverage. Maybe something to pair on?